### PR TITLE
[TOPIC-BLE-LLCP] Terminate support

### PR DESCRIPTION
**Pull request by thoh-ot**

### CODE REVIEW

**Review by sjanc — APPROVED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_connections.c [hunk 1]
```diff
@@ -390,6 +393,28 @@ void ull_conn_done(struct node_rx_event_done *done)
 		return;
 	}
 
+	/* Master transmitted ack for the received terminate ind or
```
Comment (erbr-ot) [inline subsys/bluetooth/controller/ll_sw/ull_connections.c:401]: I find this kind of if statements with interleaved #ifdefs really hard to read 
Also  (or therefore ;-)) I don't understand why the (reason == BT_HCI_ERR_TERM_DUE_TO_MIC_FAIL) check is under CENTRAL-def when comment says slave received MIC failure

Comment (thoh-ot) [reply subsys/bluetooth/controller/ll_sw/ull_connections.c:401]: > I find this kind of if statements with interleaved #ifdefs really hard to read
> Also (or therefore ;-)) I don't understand why the (reason == BT_HCI_ERR_TERM_DUE_TO_MIC_FAIL) check is under CENTRAL-def when comment says slave received MIC failure

This code is copied directly from the legacy ull_conn.c and i haven't looked into why it's written like that. This code is not related to LLCP functionality but is existing connection logic that was left out during the creation of the ull_connections.c file.

File: subsys/bluetooth/controller/ll_sw/ull_connections.c [hunk 2]
```diff
@@ -907,8 +947,89 @@ static inline void disable(uint16_t handle)
...
+
+	/* Invalidate the connection context */
+	lll->handle = 0xFFFF;
+
+	/* Demux and flush Tx PDUs that remain enqueued in thread context */
+	ull_conn_tx_demux(UINT8_MAX);
 }
 
+static void tx_ull_flush(struct ll_conn *conn)
+{
```
Comment (erbr-ot) [inline subsys/bluetooth/controller/ll_sw/ull_connections.c:1008]: Empty on purpose?

Comment (thoh-ot) [reply subsys/bluetooth/controller/ll_sw/ull_connections.c:1008]: Yes, It's updated in a later commit (https://github.com/zephyrproject-rtos/zephyr/pull/32332/commits/82abbb3714d7f9b51101616581de13886681c77c)

Comment (erbr-ot) [reply subsys/bluetooth/controller/ll_sw/ull_connections.c:1008]: I see that now - sorry :-)

**Review by erbr-ot — APPROVED** [review]
LGTM

**Pull request merged**